odc/geo/_xr_interop.py

```diff
@@ -318,6 +318,17 @@ def mask(
         value_inside=not invert,
     )
 
+    # Repeat rasterized mask across all non-spatial dims
```

Comment:
Instead of using `spatial_dims` method to get names of spatial dimensions, can you please use `xx.odc.ydim` to obtain location of the y-axis (x-axis always follows y) and expand `rasterized` using that.

```python
non_sdims = (*xx.dims[:ydim], xx.dims[ydim+2:])
```